### ISSUE TITLE
Plugin causes empty window when editing *.agda files.

### ISSUE THREAD

**Issue by mobotsar** [isovector/cornelis#38]
My setup is Fedora 35 Workstation, NVIM v0.6.1, vim-plug 0.11.0, Stack 2.7.3 x86_64, Agda 2.6.1.3. 
I installed vim-plug by running the following command (and adding the begin/end directives to my .config/nvim/init.vim).
```
sh -c 'curl -fLo "${XDG_DATA_HOME:-$HOME/.local/share}"/nvim/site/autoload/plug.vim --create-dirs \
       https://raw.githubusercontent.com/junegunn/vim-plug/master/plug.vim'
```

I installed the cornelis plugin as directed in the readme, by adding three lines to my `init.vim` between the vim-plug directives, then running `PlugInstall` from within Neovim.

```
call plug#begin()    
Plug 'kana/vim-textobj-user'    
Plug 'neovimhaskell/nvim-hs.vim'    
Plug 'isovector/cornelis'        
call plug#end()
```

Once these steps are completed, upon running `nvim foo.agda`, my terminal goes completely black and becomes unresponsive, forcing me to close the window or issue `kill -9` to return control to the shell. This does not happen with files of any other ending, e.g. foo.hs works as it did before.

It seems likely that this issue is due to some step missing in the configuration/setup of the plugin, but as far as I can tell I've followed the installation instructions to the letter.

If you need any more information, I'm more than happy to provide it.

**Comment by isovector** [comment]
I think this is caused by `nvim-hs.vim` building `cornelis` for the first time. The unresponsiveness should subside after a build is complete; alternatively, jump into the `~/.vim/plugged/cornelis` and run `stack build` there.

**Comment by mobotsar** [comment]
Okay, thanks for the quick response. Perhaps that could be indicated in the readme, so as to prevent erroneous issues like this one from being created.

I'm running the build now, and I'll let you know if it fails to solve the issue.

**Comment by mobotsar** [comment]
It does seem to have fixed _the_ issue. I'm now experiencing another issue `error in $: key "showIrrelevantArguments" not found`. So as to not hijack my own issue thread, I'm going to install in a fresh environment and see how that goes, to make sure it's not some quirk of my machine.

**Comment by isovector** [comment]
That sounds like the JSON from agda isn't what I expected it to be. It's probably a bug in cornelis. Do you have a small repro?

**Comment by isovector** [comment]
Alternatively, please post the results from `agda --interaction-json` of:

```
IOTCM "/home/your/file/here.agda" None Direct (Cmd_load "/home/your/file/here.agda" [])
```

**Comment by mobotsar** [comment]
If by repro you mean steps for reproduction, then it's essentially just the steps I listed above: adding the binding function code section from the readme to my `init.vim` and opening a valid agda file. Maybe it's a version incompatibility? I'm on Agda 2.6.1 whereas I think the current release is 2.6.3. Is there a specific peice of information you'd like that would help diagnose the problem?

**Comment by isovector** [comment]
I think it's caused by some particular json response caused by your particular agda file. Posting the results of the `IOTCM` should be enough for me to fix the bug.

**Comment by mobotsar** [comment]
I'm afraid I don't know what IOTCM is, but if you give me a file to check or command to invoke, I'll do that.

**Comment by isovector** [comment]
Try running `agda --interaction-json`, and then inside typing 

```
IOTCM "/home/your/file/here.agda" None Direct (Cmd_load "/home/your/file/here.agda" [])
```

with the path set to the file you're trying to load in cornelis. The output should be a huge json dump, and please post that here! Sorry for the all the hurdles :(

**Comment by mobotsar** [comment]
I had the file named something different from the module name by accident. Making them the same fixed that issue and introduced a new one. With the Agda file

```
module Bleed where

data N : Set where
  zero : N
  such : N -> N
```
I get `Error in $.info.errors: parsing [] failed, expected array, but encountered string`.

**Comment by mobotsar** [comment]
> Try running `agda --interaction-json`, and then inside typing
> 
> ```
> IOTCM "/home/your/file/here.agda" None Direct (Cmd_load "/home/your/file/here.agda" [])
> ```
> 
> with the path set to the file you're trying to load in cornelis. The output should be a huge json dump, and please post that here! Sorry for the all the hurdles :(

No worries.

{"status":{"checked":false,"showImplicitArguments":false},"kind":"Status"}
{"kind":"ClearRunningInfo"}
{"kind":"ClearHighlighting","tokenBased":"NotOnlyTokenBased"}
{"kind":"DisplayInfo","info":{"kind":"Error","message":"/home/al/zz/agda/lrn/Noia.agda:1,8-13\nThe name of the top level module does not match the file name. The\nmodule Bleed should be defined in one of the following files:\n  /home/al/zz/agda/lrn/Bleed.agda\n  /home/al/zz/agda/lrn/Bleed.lagda\n  /home/al/.agda/plfa/standard-library/src/Bleed.agda\n  /home/al/.agda/plfa/standard-library/src/Bleed.lagda\n  /home/al/agda/.stack-work/install/x86_64-linux-tinfo6/9b9b41e80624553baaab8948c25431f88111a6ee4b3898d93fa1657f15635d37/8.8.3/share/x86_64-linux-ghc-8.8.3/Agda-2.6.1.3/lib/prim/Bleed.agda\n  /home/al/agda/.stack-work/install/x86_64-linux-tinfo6/9b9b41e80624553baaab8948c25431f88111a6ee4b3898d93fa1657f15635d37/8.8.3/share/x86_64-linux-ghc-8.8.3/Agda-2.6.1.3/lib/prim/Bleed.lagda"}}
{"kind":"JumpToError","filepath":"/home/al/zz/agda/lrn/Noia.agda","position":8}
{"kind":"HighlightingInfo","direct":true,"info":{"remove":false,"payload":[{"atoms":["error"],"note":"/home/al/zz/agda/lrn/Noia.agda:1,8-13\nThe name of the top level module does not match the file name. The\nmodule Bleed should be defined in one of the following files:\n  /home/al/zz/agda/lrn/Bleed.agda\n  /home/al/zz/agda/lrn/Bleed.lagda\n  /home/al/.agda/plfa/standard-library/src/Bleed.agda\n  /home/al/.agda/plfa/standard-library/src/Bleed.lagda\n  /home/al/agda/.stack-work/install/x86_64-linux-tinfo6/9b9b41e80624553baaab8948c25431f88111a6ee4b3898d93fa1657f15635d37/8.8.3/share/x86_64-linux-ghc-8.8.3/Agda-2.6.1.3/lib/prim/Bleed.agda\n  /home/al/agda/.stack-work/install/x86_64-linux-tinfo6/9b9b41e80624553baaab8948c25431f88111a6ee4b3898d93fa1657f15635d37/8.8.3/share/x86_64-linux-ghc-8.8.3/Agda-2.6.1.3/lib/prim/Bleed.lagda","definitionSite":null,"range":[8,13],"tokenBased":"NotOnlyTokenBased"}]}}
{"status":{"checked":false,"showImplicitArguments":false},"kind":"Status"}

**Comment by isovector** [comment]
Oh wow, it looks like something is going horrendously wrong here. Is upgrading agda feasible? I suspect the JSON interface has changed recently.

At the very least, you'll want these two patches

https://github.com/agda/agda/pull/5776
https://github.com/agda/agda/pull/5752 (already in head)

**Comment by mobotsar** [comment]
I may have some difficulty upgrading Agda due to some environment constraints, but I'll try to get that done and report back. It could take a few hours, so don't feel obligated to wait around for me, but I'll try to be quick.

**Comment by isovector** [comment]
No worries. I'm on a call right now, but will try to patch the JSON decoder later today so it can unblock you.

**Comment by mobotsar** [comment]
I was able to upgrade to the most recent version released on hackage, and everything seems to be working correctly now.

**Comment by isovector** [comment]
Fantastic! If you ever do need to downgrade, I think I hacked in support via 1d5c0f9029d51f310862718b10f4d91bc719d9c4

**Comment by isovector** [comment]
I'll write some documentation on what we've learned today. Thanks for your patience!

**Comment by isovector** [comment]
The startup delay, and agda versioning issues are now documented as of e7f6eaea4be31748929b49a925a3a9365682e9ec.
